django-stubs/db/models/fields/files.pyi

```diff
@@ -31,6 +31,8 @@ class FieldFile(File):
     def delete(self, save: bool = ...) -> None: ...
     @property
     def closed(self) -> bool: ...
+    def __getstate__(self) -> dict[str, Any]: ...
+    def __setstate__(self, state: dict[str, Any]) -> None: ...
```

Comment:
You can also add `__eq__` and `__hash__` as methods from https://github.com/django/django/blob/617bcf611f3daa796e4054ba041089ece30a32fc/django/db/models/fields/files.py#L25-L33

and you can also add `AltersData` base class: https://github.com/django/django/blob/617bcf611f3daa796e4054ba041089ece30a32fc/django/db/models/fields/files.py#L17

While we are at it.

Reply:
Done.

Please check whether you agree with how I annotated `__eq__()`:
- I assumed `hasattr(other, "name")` was a lazy way to check for `FieldFile`, but if it is intended to accept other classes with a `name` attribute, we should define a `Protocol` instead.
- It accepts `str` for backwards compatibility, but it doesn't state that compatibility is deprecated, so I added it as part of the signature.